button-merchant/src/test/java/com/usebutton/merchant/MainThreadExecutorTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * MainThreadExecutorTest.java
+ *
+ * Copyright (c) 2018 Button, Inc. (https://usebutton.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.usebutton.merchant;
+
+import android.os.Looper;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.concurrent.Executor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+public class MainThreadExecutorTest {
+
+    private Executor executor = new MainThreadExecutor();
+
+    @Test
+    public void executor_shouldExecute() throws Exception {
+        final boolean[] executed = { false };
+
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                assertEquals(Looper.myLooper(), Looper.getMainLooper());
+                executed[0] = true;
+            }
+        });
+
+        assertTrue(executed[0]);
```

Comment:
You may want to add `Robolectric.flushForegroundThreadScheduler();` before the assertion to avoid a possible race condition.

Reply:
Good call!